snapcraft/internal/repo/snaps.py

```diff
@@ -0,0 +1,196 @@
+
+        Validity of the results are determined by checking self.installed."""
+        if self._is_installed is None:
+            try:
+                self._local_snap_info = _get_local_snap_info(self.name)
+                self._is_installed = True
+            except exceptions.HTTPError:
+                self._is_installed = False
+        return self._local_snap_info
+
+    def get_store_snap_info(self):
+        """Returns a store payload for the snap.
+
+        Validity of the results are determined by checking self.remote."""
+        if self._is_on_store is None:
+            try:
+                self._store_snap_info = _get_store_snap_info(self.name)
+                self._is_on_store = True
+            except exceptions.HTTPError:
+                self._is_on_store = False
+        return self._store_snap_info
+
+    def _get_store_channels(self):
+        snap_store_info = self.get_store_snap_info()
+        if not self.on_store:
+            return dict()
+
+        return snap_store_info['result'][0]['channels']
+
+    def get_current_channel(self):
+        current_channel = ''
+        if self.installed:
+            local_snap_info = self.get_local_snap_info()
+            current_channel = local_snap_info['result']['channel']
+            if current_channel == 'stable':
+                current_channel = 'latest/stable'
+        return current_channel
+
+    def is_classic(self):
+        store_channels = self._get_store_channels()
+        return store_channels[self.channel]['confinement'] == 'classic'
+
+    def is_valid(self):
+        """Check if the snap is valid."""
+        if not self.on_store:
+            return False
+        store_channels = self._get_store_channels()
+        return self.channel in store_channels.keys()
+
+    def install(self):
+        """Installs the snap onto the system."""
+        snap_install_cmd = []
+        if _snap_command_requires_sudo():
+            snap_install_cmd = ['sudo']
+        snap_install_cmd.extend(['snap', 'install', self.name])
+        if self._original_channel:
+            snap_install_cmd.extend(['--channel', self._original_channel])
+        if self.is_classic:
+            snap_install_cmd.extend(['--classic'])
+        try:
+            check_call(snap_install_cmd)
+        except CalledProcessError as install_error:
+            raise errors.SnapInstallError(snap_name=self.name,
+                                          snap_channel=self.channel)
+
+    def refresh(self):
+        """Refreshes a snap onto a channel on the system."""
+        snap_refresh_cmd = []
+        if _snap_command_requires_sudo():
+            snap_refresh_cmd = ['sudo']
+        snap_refresh_cmd.extend(['snap', 'refresh', self.name,
+                                 '--channel', self.channel])
+        try:
+            check_call(snap_refresh_cmd)
+        except CalledProcessError as install_error:
+            raise errors.SnapInstallError(snap_name=self.name,
```

Comment:
A Refresh error would be more appropriate here.

Reply:
got it